Mathlib/Algebra/Module/PID.lean

```diff
@@ -55,7 +53,7 @@ universe u v
 
 variable {R : Type u} [CommRing R] [IsPrincipalIdealRing R]
 variable {M : Type v} [AddCommGroup M] [Module R M]
-variable {N : Type max u v} [AddCommGroup N] [Module R N]
+variable {N : Type v} [AddCommGroup N] [Module R N]
```

Comment:
Are `N` and `M` required to live in the same universe?  Can you try changing all of them to `Type*` instead?

Reply:
Ideally there should be no `N`, only use `M` is enough. But somehow it's not working in my first try. Let me try it again.

I don't think it's easy to change to `Type*` as it seems that the original proof is sensitive to universe levels (for example, exists something in `Type u` such that blah blah).

Reply:
All `N` are removed. Please review.